ledger/sandbox/src/main/resources/db/migration/postgres/V26_0__Contracts_new_schema.sql

```diff
@@ -0,0 +1,60 @@
+-- Copyright (c) 2019 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+---------------------------------------------------------------------------------------------------
+-- V26: Contracts new schema
+--
+-- Used for interpretation and validation by the DAML engine
+---------------------------------------------------------------------------------------------------
+
+-- contains all active and divulged contracts
+create table participant_contracts
+(
+    contract_id varchar primary key not null,
+    template_id varchar not null,
+    -- create_argument needs to be nullable to leave this field empty as part of this migration
+    -- a second (java) migration will fill it up
+    -- a third (sql) migration will make this non-nullable, as it's supposed to always be populated
+    create_argument bytea,
+
+    -- the following fields are null for divulged contracts
+    create_stakeholders varchar array,
+    create_key_hash bytea,
+    create_ledger_effective_time timestamp
+);
+
+-- support looking up a contract by key
+create unique index on participant_contracts(create_key_hash);
+
+insert into participant_contracts
+select
+    contracts.id as contract_id,
+    contracts.package_id || ':' || contracts.name as template_id,
+    null as create_argument, -- filled up in a subsequent migration
+    array_agg(contract_observers.observer) || array_agg(contract_signatories.signatory) as create_stakeholders,
```

Comment:
Cool, I didn't know `||` also acts as array concatenation 👍

Reply:
```suggestion
    array_agg(contract_observers.observer) filter (where contract_observers.observer is not null) || array_agg(contract_signatories.signatory) filter (where contract_signatories.signatory is not null) as create_stakeholders,
```
This filters out the `NULL` value in case there are no observers or signatories (in case of divulged contracts).

Reply:
I'm not sure this is what we want. The stakeholders are supposed to null for divulged contracts.

Reply:
Sorry, I wasn't clear. the added filter makes it so that we don't get `{NULL, 'Alice'}` in case there are no observers.

```
select
contract_data.id as contract_id,
array_agg(contract_observers.observer) filter (where contract_observers.observer is not null) || array_agg(contract_signatories.signatory) as create_stakeholders
from contract_data
left join contract_observers on contract_observers.contract_id = contract_data.id
left join contract_signatories on contract_signatories.contract_id = contract_data.id
group by (contract_data.id);
 contract_id | create_stakeholders
-------------+---------------------
 #0:0        | {Alice}
 #1:0        | {Alice}
 #2:0        | {Alice}
 #3:0        | {Alice}
(4 rows)

select
contract_data.id as contract_id,
array_agg(contract_observers.observer) || array_agg(contract_signatories.signatory) as create_stakeholders
from contract_data
left join contract_observers on contract_observers.contract_id = contract_data.id
left join contract_signatories on contract_signatories.contract_id = contract_data.id
group by (contract_data.id);
 contract_id | create_stakeholders
-------------+---------------------
 #0:0        | {NULL,Alice}
 #1:0        | {NULL,Alice}
 #2:0        | {NULL,Alice}
 #3:0        | {NULL,Alice}
(4 rows)
```

Reply:
And in the case of divulged contracts, also the join for signatories will yield such a `NULL` array element.

Reply:
Ok, thanks, let me re-run the tests one last time.

Reply:
Tests pass! :tada: